src/allmydata/test/web/test_web.py

```diff
@@ -1721,26 +1741,26 @@ def test_PUT_NEWFILEURL_unlinked_mdmf(self):
         d = self.PUT("/uri?format=mdmf",
                      contents)
         def _got_filecap(filecap):
-            self.failUnless(filecap.startswith("URI:MDMF"))
+            self.failUnless(filecap.startswith(b"URI:MDMF"))
             return filecap
         d.addCallback(_got_filecap)
-        d.addCallback(lambda filecap: self.GET("/uri/%s?t=json" % filecap))
-        d.addCallback(lambda json: self.failUnlessIn("MDMF", json))
+        d.addCallback(lambda filecap: self.GET("/uri/%s?t=json" % str(filecap, "utf-8")))
+        d.addCallback(lambda json: self.failUnlessIn(b"MDMF", json))
         return d
 
     def test_PUT_NEWFILEURL_unlinked_sdmf(self):
         contents = self.NEWFILE_CONTENTS * 300000
         d = self.PUT("/uri?format=sdmf",
                      contents)
-        d.addCallback(lambda filecap: self.GET("/uri/%s?t=json" % filecap))
-        d.addCallback(lambda json: self.failUnlessIn("SDMF", json))
+        d.addCallback(lambda filecap: self.GET("/uri/%s?t=json" % str(filecap, "utf-8")))
+        d.addCallback(lambda json: self.failUnlessIn(b"SDMF", json))
         return d
 
     @inlineCallbacks
     def test_PUT_NEWFILEURL_unlinked_bad_format(self):
         contents = self.NEWFILE_CONTENTS * 300000
         yield self.assertHTTPError(self.webish_url + "/uri?format=foo", 400,
-                                   "Unknown format: foo",
+                                   "Unknown format:",
                                    method="put", data=contents)
```

Comment:
What happened to the "foo"?  Did it turn in to `b"foo"` or something?

I guess it would be nice to continue to make some kind of assertion about "foo" in this string... although even more interesting would be a test for `format=<some csrf string>` which demonstrates it gets quoted correctly (but clearly this is beyond the scope of just porting).

Reply:
Yeah, it turns into `b"foo"` and I ran out of spoons to fix all those. Which is maybe a bad reason not to do it. So I guess I'll go undo this and other similar changes and fix code.